lighthouse-core/scripts/pptr-run-devtools.js

```diff
@@ -152,122 +176,146 @@ const startLighthouse = `
- * @return {Promise<{lhr: LH.Result, artifacts: LH.Artifacts}>}
+ * @param {string[]=} chromeFlags
+ * @return {Promise<{lhr: LH.Result, artifacts: LH.Artifacts, logs: string[]}>}
  */
-async function testPage(page, browser, url, config) {
-  const targets = await browser.targets();
-  const inspectorTarget = targets.filter(t => t.url().includes('devtools'))[1];
-  if (!inspectorTarget) throw new Error('No inspector found.');
-
-  const session = await inspectorTarget.createCDPSession();
-  await session.send('Runtime.enable');
-
-  // Navigate to page and wait for initial HTML to be parsed before trying to start LH.
-  await new Promise((resolve, reject) => {
-    page.target().createCDPSession()
-      .then(session => {
-        session.send('Page.enable').then(() => {
-          session.once('Page.domContentEventFired', resolve);
-          page.goto(url);
-        });
-      })
-      .catch(reject);
+async function testUrlFromDevtools(url, config, chromeFlags) {
+  const browser = await puppeteer.launch({
+    executablePath: getChromePath(),
+    args: chromeFlags,
+    devtools: true,
   });
 
-  if (config) {
-    // Must attach to the Lighthouse worker target and override the `self.createConfig`
-    // function, allowing us to use any config we want.
-    session.send('Target.setAutoAttach', {
-      autoAttach: true, flatten: true, waitForDebuggerOnStart: false,
-    });
-    session.once('Target.attachedToTarget', async (event) => {
-      if (event.targetInfo.type !== 'worker') throw new Error('expected lighthouse worker');
-
-      const targets = await browser.targets();
-      const workerTarget = targets.find(t => t._targetId === event.targetInfo.targetId);
-      if (!workerTarget) throw new Error('No lighthouse worker target found.');
-
-      const workerSession = await workerTarget.createCDPSession();
-      await Promise.all([
-        workerSession.send('Runtime.enable'),
-        workerSession.send('Debugger.enable'),
-        new Promise(resolve => {
-          workerSession.once('Debugger.scriptParsed', resolve);
-        }),
-      ]);
-      await workerSession.send('Debugger.pause');
-      await workerSession.send('Runtime.evaluate', {
-        expression: `self.createConfig = () => (${JSON.stringify(config)});`,
-      });
-      await workerSession.send('Debugger.resume');
-    });
-  }
+  try {
+    if ((await browser.version()).startsWith('Headless')) {
+      throw new Error('You cannot use headless');
+    }
 
-  /** @type {Omit<LH.Puppeteer.Protocol.Runtime.EvaluateResponse, 'result'>|undefined} */
-  let startLHResponse;
-  while (!startLHResponse || startLHResponse.exceptionDetails) {
-    if (startLHResponse) await new Promise(resolve => setTimeout(resolve, 1000));
-    startLHResponse = await session.send('Runtime.evaluate', {
-      expression: startLighthouse,
-      awaitPromise: true,
-    }).catch(err => ({exceptionDetails: err}));
-  }
+    const page = (await browser.pages())[0];
```

Comment:
This is subtle but very important. We were previously creating a new tab after the browser starts. This would sometimes lead to a situation where the new tab is not being focused during the test.

To prevent this, we can just use the default tab that the browser starts with by default.